app/scripts/modules/core/src/pipeline/config/triggers/PipelineRoles.tsx

```diff
@@ -0,0 +1,45 @@
+import * as React from 'react';
+import { useState, useEffect } from 'react';
+import Select, { Option } from 'react-select';
+
+import { AuthenticationService } from 'core/authentication';
+import { HelpField } from 'core/help';
+
+export interface IPipelineRolesConfigProps {
+  roles: any[];
+  updateRoles: (roles: string[]) => void;
+}
+
+export const PipelineRoles = (props: IPipelineRolesConfigProps) => {
+  const [allowedRoles, setAllowedRoles] = useState([]);
+
+  useEffect(() => {
+    setAllowedRoles(AuthenticationService.getAuthenticatedUser().roles);
```

Comment:
If my understanding is correct, `useEffect` will be called after each render -- do we need to fetch the allowed roles repeatedly? I only ask because it appears that in the controller we're replacing, this is only called on init.

Reply:
Oh, perhaps we could use this pattern to only conditionally call `useEffect`: https://reactjs.org/docs/hooks-reference.html#conditionally-firing-an-effect
I also might be misunderstanding, just trying to read up on this stuff recently!

Reply:
Good Idea, I set the conditional to an empty array which should only trigger once.